plugin/dashboard.lua

```diff
@@ -33,6 +34,20 @@ api.nvim_create_autocmd('FileType', {
   end,
 })
 
+if db.session_auto_save_on_exit then
+  local session_auto_save = api.nvim_create_augroup('session_auto_save', { clear = true })
+
+  api.nvim_create_autocmd('VimLeavePre', {
+    group = session_auto_save,
+    callback = function()
+      pcall(vim.cmd, "NvimTreeClose")
```

Comment:
remove this line. it should write in user script.

Reply:
Since the user's autocommand would not necessarily be executed before `session_save` in order to properly remove the NvimTree buffer before saving the session I put this in here. But I have a better idea. I will introduce two cusotomizable events that users can add their own functions to execute before and after auto-saving. :)

Reply:
provide a autocmd events like `User DBSessionSavePre`

Reply:
I was going to give users two new fields i.e. `db.session_auto_save_pre` and `db.session_auto_save_after` which the users could assign functions to.

Reply:
I implemented this now and added a description to the README.